mongo/integration/sessions_test.go

```diff
@@ -205,71 +285,91 @@ func TestSessions(t *testing.T) {
 
-		doc := bson.D{{"x", 1}}
-		_, err := mt.Coll.InsertOne(context.Background(), doc)
-		assert.Nil(mt, err, "InsertOne error: %v", err)
-		_, err = mt.Coll.InsertOne(context.Background(), doc)
-		assert.Nil(mt, err, "InsertOne error: %v", err)
+	mt.Run("8 client side cursor that exhausts the results on the initial query immediately returns the implicit session to the pool",
+		func(mt *mtest.T) {
+			// implicit sessions are returned to the server session pool
+
+			doc := bson.D{{"x", 1}}
+			_, err := mt.Coll.InsertOne(context.Background(), doc)
+			assert.Nil(mt, err, "InsertOne error: %v", err)
+			_, err = mt.Coll.InsertOne(context.Background(), doc)
+			assert.Nil(mt, err, "InsertOne error: %v", err)
+
+			// create a cursor that will hold onto an implicit session and record the sent session ID
+			mt.ClearEvents()
+			cursor, err := mt.Coll.Find(context.Background(), bson.D{})
+			assert.Nil(mt, err, "Find error: %v", err)
+			findID := extractSentSessionID(mt)
+			assert.True(mt, cursor.Next(context.Background()), "expected Next true, got false")
+
+			// execute another operation and verify the find session ID was reused
+			_, err = mt.Coll.DeleteOne(context.Background(), bson.D{})
+			assert.Nil(mt, err, "DeleteOne error: %v", err)
+			deleteID := extractSentSessionID(mt)
+			assert.Equal(mt, findID, deleteID, "expected session ID %v, got %v", findID, deleteID)
+		})
 
-		// create a cursor that will hold onto an implicit session and record the sent session ID
-		mt.ClearEvents()
-		cursor, err := mt.Coll.Find(context.Background(), bson.D{})
-		assert.Nil(mt, err, "Find error: %v", err)
-		findID := extractSentSessionID(mt)
-		assert.True(mt, cursor.Next(context.Background()), "expected Next true, got false")
+	mt.Run("9 client side cursor that exhausts the results after a getMore immediately returns the implicit session to the pool",
+		func(mt *mtest.T) {
+			fmt.Println("start")
+			// Client-side cursor that exhausts the results after a getMore immediately returns the implicit session to the pool.
 
-		// execute another operation and verify the find session ID was reused
-		_, err = mt.Coll.DeleteOne(context.Background(), bson.D{})
-		assert.Nil(mt, err, "DeleteOne error: %v", err)
-		deleteID := extractSentSessionID(mt)
-		assert.Equal(mt, findID, deleteID, "expected session ID %v, got %v", findID, deleteID)
-	})
-	mt.Run("implicit session returned from getMore", func(mt *mtest.T) {
-		// Client-side cursor that exhausts the results after a getMore immediately returns the implicit session to the pool.
+			var docs []interface{}
+			for i := 0; i < 5; i++ {
+				docs = append(docs, bson.D{{"x", i}})
+			}
 
-		var docs []interface{}
-		for i := 0; i < 5; i++ {
-			docs = append(docs, bson.D{{"x", i}})
-		}
-		_, err := mt.Coll.InsertMany(context.Background(), docs)
-		assert.Nil(mt, err, "InsertMany error: %v", err)
+			fmt.Println("before")
+			fmt.Println("coll: ", mt.Coll)
```

Comment:
Optional: Is keeping this logging intentional? If so, consider using `mt.Log` instead.